src/network/grpc/client.rs

```diff
@@ -80,12 +76,38 @@ fn subscribe(
                     unimplemented!()
                 }
             });
-            subscription::process_block_announcements(node_id, block_sub, block_box);
+            subscription::process_block_announcements(node_id, block_sub, block_box.clone());
             subscription::process_gossip(gossip_sub, global_state);
+            process_block_solicitations(client, &mut prop_handles, block_box);
             Ok((node_id, prop_handles))
         })
 }
 
+fn process_block_solicitations(
+    mut client: Connection<BlockConfig, TcpStream, DefaultExecutor>,
+    prop_handles: &mut propagate::PeerHandles,
+    block_box: MessageBox<BlockMsg>,
+) {
+    tokio::spawn(
+        prop_handles
+            .solicit_blocks
+            .subscribe()
+            .for_each(move |block_ids| {
+                let block_box = block_box.clone();
+                client.get_blocks(&block_ids).and_then(move |blocks| {
+                    let mut block_box = block_box.clone();
+                    blocks.for_each(move |block| {
+                        block_box.send(BlockMsg::NetworkBlock(block));
+                        Ok(())
+                    })
+                })
+            })
+            .map_err(|e| {
+                info!("block solicitation failed: {:?}", e);
```

Comment:
a `warn` instead maybe? This is not really an error, i agree. But not being able to solicit a block might be problematic, and this is a warning to something that may become a problem down the line (not being able to sync the blockchain with the rest of the world).
Better having this already well sorted.